add setuptools, doc url, description to conda recipe; bump version; update SHA256

Timothy-W-Hilton opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->
<code>@<space/>conda-forge-admin, please rerender</code>

Timothy-W-Hilton committed:
add setuptools, doc url, description to conda recipe

Timothy-W-Hilton committed:
reset build number to 0

Conda-forge pull request instructions require this.

Timothy-W-Hilton committed:
bump version; update SHA256

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

Timothy-W-Hilton committed:
add rtree dependency to meta.yaml recipe

- that's in the pyproject.toml file, but missing from the conda recipe somehow.
